linq4j/src/main/java/org/apache/calcite/linq4j/tree/AbstractNode.java

```diff
@@ -71,8 +71,12 @@ public Node accept(Shuttle shuttle) {
   }
 
   public Object evaluate(Evaluator evaluator) {
-    throw new RuntimeException(
-        "evaluation not supported: " + getClass() + ":" + nodeType);
+    return evaluator.evaluate(this);
+  }
+
+  public Object evaluate() {
```

Comment:
Could you please change the title and the commit message to make it clearer? Such as `Implement evaluate method of AbstractNode`?

Reply:
changed the title and the commit message.